roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -85,10 +87,32 @@ impl ParseClientJobDeclarationMessages for JobDeclaratorDownstream {
 
             for (i, sid) in short_hash_list.iter().enumerate() {
                 let sid_: [u8; 6] = sid.to_vec().try_into().unwrap();
-                if let Some(tx_data) = short_id_mempool.get(&sid_) {
-                    txs_in_job.push(tx_data.clone());
-                } else {
-                    missing_txs.push(i as u16);
+                match short_id_mempool.get(&sid_) {
+                    Some(tx_data) => match &tx_data.tx {
+                        Some(tx) => txs_in_job.push(tx.clone()),
+                        None => {
+                            let new_tx_data = self
+                                .mempool
+                                .safe_lock(|x| x.get_client())
+                                .unwrap()
+                                .unwrap()
+                                .get_raw_transaction(&tx_data.id.to_string(), None);
+                            match new_tx_data {
+                                Ok(new_tx) => {
+                                    mempool::JDsMempool::add_tx_data_to_mempool(
+                                        self.mempool.clone(),
+                                        tx_data.clone().id,
+                                        Some(new_tx.clone()),
+                                    );
+                                    txs_in_job.push(new_tx);
+                                }
+                                Err(_) => {
+                                    missing_txs.push(i as u16);
+                                }
+                            }
+                        }
+                    },
+                    None => missing_txs.push(i as u16),
```

Comment:
Why you do not try to get the tx data from the node, before adding it to missing_txs?

Reply:
Becuase this path is the one taken when the txid is not present into the current mempool, so it means that tx is not present at all.